### PR TITLE
[!!!][FEATURE] Switch to symfony to execute commands on remote

### PR DESCRIPTION
**Pull request by chriwo**
The phpseclib packages requires a public key file to get the content and authenticate with the remote server. I use ddev for local development and since version 1.5.0 is it possible to use `ddev auth ssh` (https://github.com/drud/ddev/pull/1295). With this feature the ssh key would be added to the ssh-agent and it's not possible to get content of ssh key.

Symfony solve this problem and use the ssh-agent to get the key and authenticate with the remote server.

Also implement feature is the configuration of the temporary local path for the rsync-file. The synchronization fails in ddev because there are no write permissions in sys_get_temp_dir. Now is it possible to configure this path with e.g. `$reverseDeploy->setLocalTempPath('/var/www/html/')`

### CODE REVIEW

File: JoRo/Typo3ReverseDeployment.php [hunk 1]
```diff
@@ -443,44 +451,114 @@ public function setPathToConsoleExecutable($pathToConsoleExecutable)
...
+     * Set remote server uri
+     * @param string $remoteServer
+     */
+    public function setRemoteServer($remoteServer)
+    {
+        $this->remoteServer = $remoteServer;
+    }
+
+    /**
+     * Get remote server uir
```
Comment (ochorocho) [inline JoRo/Typo3ReverseDeployment.php]: Should be "... server uri"

**Comment by ochorocho** [comment]
Hey, thanks for your contribution. Seem like your rewrote the entire thing. 👍 
I really think it's finally time to write tests for this. As soon as i get the time I'll test and merge this.

**Comment by chriwo** [comment]
Yes, writing tests is a good idea. I can see how I can do it.

File: composer.json [hunk 2]
```diff
@@ -3,8 +3,9 @@
   "description": "Export DB and files to your TYPO3 dev installation",
   "license": "GPL-3.0-or-later",
   "require": {
-    "phpseclib/phpseclib": "^2.0",
-    "neos/utility-files": "^3.0"
```
Comment (ochorocho) [inline composer.json:7]: "neos/utility-files" is required, i'll install it after merge again

**Comment by ochorocho** [comment]
Looks good to me so far. Thanks @chriwo

**Pull request merged**